server/api.js

```diff
@@ -6,146 +6,80 @@ import logger from "./utils/logger";
 
 const router = Router();
 
-let fakeLaptopAssign = [
-	{
-		laptopId: 1,
-		laptopName: "Dell",
-		deliveryOption: "pickup",
-		address: "donator address: 123 main st",
-	},
-	{
-		laptopId: 2,
-		laptopName: "Mac",
-		deliveryOption: "ship",
-		address: "donator address: 124 main st",
-	},
-];
-
-let fakeRequests = [
-	{
-		id: 100,
-		firstName: "bob",
-		lastName: "the builder",
-		email: "email@email.com",
-		phoneNumber: "073820384924",
-		status: "WAITING",
-		laptopAssignment: {},
-		requestAddress: "",
-	},
-	{
-		id: 110,
-		firstName: "hulk",
-		lastName: "the smasher",
-		email: "email@email.com",
-		phoneNumber: "073820384924",
-		status: "ASSIGNED",
-		laptopAssignment: fakeLaptopAssign[1],
-		requestAddress: " users address : 123 south street",
-	},
-	{
-		id: 111,
-		firstName: "tony",
-		lastName: "stack",
-		email: "email@email.com",
-		phoneNumber: "073820384924",
-		status: "ASSIGNED",
-		laptopAssignment: fakeLaptopAssign[0],
-		requestAddress: "users address : 123 south street",
-	},
-	{
-		id: 120,
-		firstName: "silver",
-		lastName: "surfer",
-		email: "email@email.com",
-		phoneNumber: "073820384924",
-		status: "ACCEPTED",
-		laptopAssignment: fakeLaptopAssign[1],
-		requestAddress: "users address : 123 south street",
-	},
-
-	{
-		id: 121,
-		firstName: "silver",
-		lastName: "surfer",
-		email: "email@email.com",
-		phoneNumber: "073820384924",
-		status: "ACCEPTED",
-		laptopAssignment: fakeLaptopAssign[0],
-		requestAddress: "users address : 123 south street",
-	},
-
-	{
-		id: 130,
-		firstName: "hulk",
-		lastName: "the smasher",
-		email: "email@email.com",
-		phoneNumber: "073820384924",
-		status: "CANCELLED",
-		laptopAssignment: {},
-		requestAddress: "",
-	},
-	{
-		id: 140,
-		firstName: "Don",
-		lastName: "Yen",
-		email: "email@email.com",
-		phoneNumber: "073820384924",
-		status: "FULLFILLED",
-	},
-];
-
 router.get("/laptop_donation/:id", async (req, res) => {
 	try {
 		const result = await db.query(
 			"SELECT * from laptop_donation WHERE id = $1",
 			[req.params.id]
 		);
-		let id = result.rows[0].id;
-		let name = result.rows[0].name;
-		let address = result.rows[0].address;
-		let numberOfLaptops = result.rows[0].number_of_laptops;
-		let phoneNumber = result.rows[0].phone_number;
-		let email = result.rows[0].email;
-		let deliveryOption = result.rows[0].delivery_option;
+
 		let laptopDonation = {
-			id: id,
-			name: name,
-			address: address,
-			numberOfLaptops: numberOfLaptops,
-			phoneNumber: phoneNumber,
-			email: email,
-			deliveryOption: deliveryOption,
+			id: result.rows[0].id,
```

Comment:
I would set the id here to result.rows[0].uuid

In this way, the client never knows about database ids, which is a good thing

Reply:
Done this